package com.learning.shared.exception;

import java.io.PrintWriter;
import java.io.StringWriter;

import org.apache.log4j.Logger;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestController;

import com.learning.shared.bean.ResponseBean;
import com.learning.shared.constant.StatusCode;
import com.learning.shared.util.JsonUtil;

@RestController
@ControllerAdvice
public class GlobalExceptionHandler {
	
	
	private final static Logger logger = Logger.getLogger(GlobalExceptionHandler.class);
	
	@ExceptionHandler(NullPointerException.class)
	public ResponseBean<String> nullPointerException(NullPointerException exception) {
		StringWriter sw = new StringWriter();
		exception.printStackTrace(new PrintWriter(sw));
        String exceptionAsString = sw.toString();
        System.out.println(exceptionAsString);
		logger.error(exceptionAsString);
		ResponseBean<String> responseBean = JsonUtil.createResponseBean(StatusCode.INTERNAL_SERVER_ERROR,"Somthing went wrong please try again!!",null);
		return responseBean;
	}
}
